lua/html_embed_hypothesis.lua

```diff
@@ -0,0 +1,107 @@
+local http = require "resty.http"
+
+local original_scheme = ngx.var.http_x_forwarded_proto or ngx.var.scheme;
+local query_params = ngx.req.get_query_args()
+
+local hypothesis_header = [[
+<script>
```

Comment:
This is awkward cause it'd be nice to use a templater for this but AFAIK the templater can only render the template as a response and there isn't a way to create the template as a string to pass it back to nginx.

Reply:
What are you referring to by "the templater"?

Can you use general-purpose lua packages in this context? If so perhaps a lua implementation of [Mustache templates](https://mustache.github.io) might be a solution?

Reply:
Just gave this a test run locally and it looks like that would work very nicely here. I don't necessarily want to use multiple template engines though. Ideally we would use one that allows both rendering the template as an nginx response and one that allows rendering it as a string and both would allow us to point to a template file. I'm not sure if it's worth adding another dependency to essentially do what we're doing here with string replacement. I think this requires some more research.